Core/Src/main.c

```diff
@@ -0,0 +1,229 @@
+  */
+/* USER CODE END Header */
+/* Includes ------------------------------------------------------------------*/
+#include "main.h"
+#include "spi.h"
+#include "tim.h"
+#include "usart.h"
+#include "gpio.h"
+
+/* Private includes ----------------------------------------------------------*/
+/* USER CODE BEGIN Includes */
+
+/* USER CODE END Includes */
+
+/* Private typedef -----------------------------------------------------------*/
+/* USER CODE BEGIN PTD */
+
+/* USER CODE END PTD */
+
+/* Private define ------------------------------------------------------------*/
+/* USER CODE BEGIN PD */
+/* USER CODE END PD */
+
+/* Private macro -------------------------------------------------------------*/
+/* USER CODE BEGIN PM */
+
+/* USER CODE END PM */
+
+/* Private variables ---------------------------------------------------------*/
+
+/* USER CODE BEGIN PV */
+
+/* USER CODE END PV */
+
+/* Private function prototypes -----------------------------------------------*/
+void SystemClock_Config(void);
+/* USER CODE BEGIN PFP */
+
+/* USER CODE END PFP */
+
+/* Private user code ---------------------------------------------------------*/
+/* USER CODE BEGIN 0 */
+
+/* USER CODE END 0 */
+
+/**
+  * @brief  The application entry point.
+  * @retval int
+  */
+int main(void)
+{
+  /* USER CODE BEGIN 1 */
+
+
+  /* USER CODE END 1 */
+
+  /* MCU Configuration--------------------------------------------------------*/
+
+  /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
+  HAL_Init();
+
+  /* USER CODE BEGIN Init */
+
+  /* USER CODE END Init */
+
+  /* Configure the system clock */
+  SystemClock_Config();
+
+  /* USER CODE BEGIN SysInit */
+
+  /* USER CODE END SysInit */
+
+  /* Initialize all configured peripherals */
+  MX_GPIO_Init();
+  MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
+  /* USER CODE BEGIN 2 */
+
+
+
+  // Start using PWM
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
+
+  // set PWM comparison to the lowest duty cycle
+  __HAL_TIM_SET_COMPARE(&htim1, TIM_CHANNEL_1, 32000);
+
+  /* USER CODE END 2 */
+
+  /* Infinite loop */
+  /* USER CODE BEGIN WHILE */
+  while (1)
+  {
+    /* USER CODE END WHILE */
+
+	  // set CS to low
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+
+	  // array to hold the bits to transmit to the adc
+	  //                         start bit , use CHO   , don't care
+	  uint8_t transmission[3] = {0b00000001, 0b10000000, 0b00000000};
+
+	  // array to hold the received bits from the adc
+	  uint8_t reception[3];
+
+	  // initialize full duplex mode for SPI
+	  HAL_SPI_TransmitReceive(&hspi1, transmission, reception, 1, HAL_MAX_DELAY);
+
+	  // set CS to low
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+
+
+
+
+	  // take last 10 bits and store it as an integer
+	  uint16_t adc_meaningful_data = ((reception[1] & 0b011) << 8 | reception[2]);
+
+	  // store the largest 10 bit number (1024 - 1)
+	  float largest_10_bit = 1023;
```

Comment:
this doesn't look like it is modified, consider making it a const and using the ALL_CAPS naming convention. Also although the compiler does the type conversion from integer to float for you, it is always good practice to explicitly do it your self:
`const float LARGEST_10_BIT = 1023.0f`